python/lsst/ap/association/utils.py

```diff
@@ -219,3 +222,66 @@ def make_empty_catalog(apdbSchema, tableName):
         for columnDef in table.columns
     }
     return pd.DataFrame(data)
+
+
+def getMidpointFromTimespan(timespan, allowEndpoints=True):
+    """Safely retrieve the midpoint in TAI from a Timespan.
+
+    Parameters
+    ----------
+    timespan : `lsst.daf.butler.Timespan` of `astropy.time.Time`
+        A Timespan centered on the midpoint of a visit.
+    allowEndpoints : `bool`, optional
+        If set, return the start or end of an open timespan.
+
+    Returns
+    -------
+    midpoint : `astropy.time.Time`
+        The midpoint of the timespan.
+
+    Raises
+    ------
+    AttributeError
+        Raised if either the start or end of the timespan is None, and
+        ``allowEndpoints`` is not set.
+    ValueError
+        Raised if the timespan is empty.
+    """
+    if (timespan.begin == Timespan.EMPTY) or (timespan.begin == Timespan.EMPTY):
+        raise ValueError("Cannot compute midpoint: EMPTY Timespan.")
+
+    try:
+        interval = timespan.end.tai - timespan.begin.tai
+        return timespan.begin.tai + interval/2
```

Comment:
One extra request: could you move the conversion to TAI out of this function to make it more general-purpose? That would let Jake Kurlander use it on #227.